launcher/src/main/java/org/apache/spark/launcher/SparkSubmitCommandBuilder.java

```diff
@@ -339,6 +339,16 @@ private void checkJavaOptions(String javaOptions) {
     }
   }
 
+  private String sanitizeConfValue(String value) {
+    if (value != null) {
+      String[] unsafeChars = {"`", "$(", ")", ";", "&", "|", "<", ">", "*", "?"};
```

Comment:
It'd be nicer if we have any reference to justify this specification

Reply:
@HyukjinKwon Thank you for your feedback! I've added references and comments to justify the specification of the unsafe characters in the sanitizeConfValue method.